### PR TITLE
File Input Documentation

### PR DESCRIPTION
**Pull request by gewenyu99**
## What does this PR do?

We have a gap in our storage docs. After introducing the `InputFile` class in many SDKs, we don't document this behavior clearly. This will be added, and the `createFile` endpoints will have updated descriptions that point here.

@Meldiron @abnegate @lohanidamodar Please help me fill in the `InputFile` implementations you worked on.

If no `InputFile` class is available, you can document some examples like this.
Screenshot 2023-02-21 at 5 46 18 PM

If `InputFile` is available with multiple ways to use, you can use a table:
Screenshot 2023-02-21 at 5 49 21 PM

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

**Comment by abnegate** [comment]
@gewenyu99 

**Apple/Swift**
`InputFile.fromPath(_ path: String)`: Used to upload files from a provided path.
`InputFile.fromData(_ data: Data, filename: String, mimeType: String)`: Used to upload files from a [Data](https://developer.apple.com/documentation/foundation/data) object.
`InputFile.fromBuffer(_ buffer: ByteBuffer, filename: String, mimeType: String)`: Used to upload files from a [NIO ByteBuffer](https://swiftinit.org/reference/swift-nio/niocore/bytebuffer) object.

**Kotlin/Android:**
`InputFile.fromPath(path: String)`: Used to upload files from a provided path.
`InputFile.fromFile(file: File)`: Used to upload files from a provided [File](https://docs.oracle.com/javase/8/docs/api/java/io/File.html) object.
`InputFile.fromBytes(bytes: ByteArray, filename: String, mimeType: String)`: Used to upload files from a [ByteArray](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-byte-array/) object.

**Ruby:**
`Input.from_path(path)`: Used to upload files from a provided path.
`Input.from_string(string)`: Used to upload files from string content.
`Input.from_bytes(bytes)`: Used to upload files from an array of bytes.

**Python:**
`Input.from_path(path)`: Used to upload files from a provided path.
`Input.from_bytes(bytes)`: Used to upload files from an array of bytes.

**Comment by gewenyu99** [comment]
@Meldiron @lohanidamodar Can you provide some information about the InputFile classes you implemented ? 😃

**Comment by Meldiron** [comment]
Rest - no changes
GraphQL - no changes
CLI - no changes
Web: No changes (we probably should implement)
Dart: constructor with `path` or `bytes` ( https://github.com/appwrite/sdk-for-dart/blob/master/lib/src/input_file.dart#L11 )
Flutter: Same as Dart ( https://github.com/appwrite/sdk-for-flutter/blob/master/lib/src/input_file.dart#L11 )
Deno: `fromPath() fromBuffer() fromPlainText()` ( https://github.com/appwrite/sdk-for-deno/blob/master/src/inputFile.ts#L15-L33 )
Node: `fromPath(), fromBuffer(), fromBlob(), fromStream(), fromPlainText()` ( https://github.com/appwrite/sdk-for-node/blob/master/lib/inputFile.js#L10-L38 )
PHP:  `withPath() and withData()` ( https://github.com/appwrite/sdk-for-php/blob/master/src/Appwrite/InputFile.php#L32-L50 )

**Comment by lohanidamodar** [comment]
@gewenyu99 @Meldiron @Meldiron I'm thinking of deprecating the constructors that accepts either path or bytes and instead introduce `InputFile.fromBytes` and `InputFile.fromPath` just like that of swift and android?

File: app/views/docs/storage.phtml [hunk 1]
```diff
@@ -181,9 +181,345 @@ File content.
...
+    <table class="full text-size-small">
+        <thead>
+        <tr>
+            <td style="width: 200px">Method</td>
+            <td>Description</td>
+        </tr>
+        </thead>
+        <tbody>
+        <tr>
+            <td><code>InputFile.fromPath(path: [PATH], filename: [NAME], contentType: [MIME TYPE])</code></td>
```
Comment (lohanidamodar) [inline app/views/docs/storage.phtml:228]: For path, filename and content type are optional

File: app/views/docs/storage.phtml [hunk 2]
```diff
@@ -181,9 +181,345 @@ File content.
...
+            <td>Used to upload files from a provided path. Used for Flutter apps on mobile and desktop.</td>
+        </tr>
+        <tr>
+            <td><code>InputFile.fromBytes(filename: [NAME], contentType: [MIME TYPE], bytes: [BYTE LIST])</code></td>
```
Comment (lohanidamodar) [inline app/views/docs/storage.phtml:232]: for bytes filename and bytes are required (can we do bytes first, filename second, and content type last ) mention content-type to be optional

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Left couple comments for Flutter.

**Review by lohanidamodar — APPROVED** [review]
Flutter looks good.

**Review by abnegate — APPROVED** [review]

File: app/views/docs/storage.phtml [hunk 3]
```diff
@@ -181,9 +181,345 @@ File content.
...
+<h3><a href="/docs/storage#file-input-client" id="file-input-client">Client SDK</a></h3>
+<ul class="phases clear" data-ui-phases>
+<li>
+    <h3>Web</h3>
+    <p>
+        The Appwrite Web SDK expects a <a href="https://developer.mozilla.org/en-US/docs/Web/API/File" target="_blank" rel="noopener">File</a> object for file creation. This is most commonly associated with DOM file inputs.
+    </p>
+
+    <p>
+        For example, for the input tag <code><?php echo htmlentities('<input type="file" id="uploader" multiple />'); ?></code>, you would call create file like this:
```
Comment (Meldiron) [inline app/views/docs/storage.phtml:203]:
```suggestion
        For example, for the input tag <code><?php echo htmlentities('<input type="file" id="uploader" />'); ?></code>, you would call create file like this:
```

I would remove `multiple`. That way browser only let's you pick one, so doing `[0]` later is cool.

If they want to upload multiple files, they would need for loop.

**Review by Meldiron — APPROVED** [review]

**Pull request merged**